design/proposals/encoded_token_response.md

```diff
@@ -0,0 +1,148 @@
+# Retrieve a Conjur secret using the authn-azure Conjur access token
+secret=$(curl -k -H "Authorization: Token token=\"$authn_azure_access_token\"" \
+  https://"$CONJUR_SERVER_DNS":8443/secrets/cucumber/variable/secrets/test-variable)
+
+echo "Retrieved secret ${secret} from Conjur!!!"
+```
+[Code reference](https://github.com/cyberark/conjur/blob/application_identity_validation/ci/authn-azure/run-authn-azure.sh#L16)
+
+
+The requested situation is to support both response of current **Flattened** JWS JSON Serialization with content type `application/json` and encoded JWS **Compact** Serialization with content type `text\plain`. 
+
+## Issue description
+1. HTTP server best practices: Using `application\json` as the content type to encoded output (not a JSON) is bad practice.
+    So encoded output shouldn't use `application/json` unless it will be in JSON format. 
+    The `../autheticate` response is in JSON format so it makes sense to use `application\json` as the response content type.
+        As well as `../logic`,  which returns the api key as `text\plain`.
+
+## Solution - Accept HTTP Header
+According to [MDN web docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept):
+ > The `Accept` request HTTP header advertises which content types the client is able to understand. Using content negotiation, the server then selects one of the proposals, uses it and informs the client of its choice with the `Content-Type` response header.
+
+To conclude, we can choose the response's `Content-Type` by the request's `Accept` header.
+
+### Design
+When `../authenticate` request the response will be encoded **only** if the `Accept` header is `text/plain`. 
+Otherwise, returns json access token as `application/json`.
+
+```ruby
+def authenticate
+    # get the authn access token
+    content_type = (request.headers["Accept"] == "text\plain") ? :text : :json
+    # encode Base64 if needed
+    render content_type => auth_token
+```
+### Backwards compatibility
+Default behaviour: Only if the `Accept` is set to `text\plain` the response will be encoded.
+
+## Solution - URL Patten
+According to [Rails Routing Guides](https://guides.rubyonrails.org/routing.html#route-globbing-and-wildcard-segments)  we can use wildcards to suggests the requested `Content_type`.
+
+By requesting `../authenticate.encoded` the output will be encoded access token.    
+
+### Design
+When `../authenticate` request the response will be encoded **only** if the url suffix (wildcard) is `.encoded`.
+Otherwise, returns json access token as `application/json`
+
+```ruby
+def authenticate
+    # get the authn access token
+    # extract the url suffix
+    content_type = ( url_suffix == ".encoded") ? :text : :json
+    # encode Base64 if needed
+    render content_type => auth_token
+```
+### Backwards compatibility
+Default behaviour: If the request doesn't include the `encoded` wildcard explicitly the response will be decoded.
```

Comment:
Current behavior is that `authenticate.*` returns 404

Reply:
Added